### PR TITLE
Add possibility to define group id manually for ZHA

### PR DESCRIPTION
**Pull request by tsydd**
## Proposed change

Backend already supports manual group ids. The PR adds the feature to the frontend

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by dmulcahey** [comment]
Ugh this is an awful way to implement this at the device level 😞. Thanks for the reply!

File: src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts [hunk 1]
```diff
@@ -75,6 +78,15 @@ export class ZHAAddGroupPage extends LitElement {
             )}
           ></paper-input>
 
+          <paper-input
+            type="number"
+            .value=${this._groupId}
+            @value-changed=${this._handleGroupIdChange}
+            placeholder=${this.hass!.localize(
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts:85]:
```suggestion
            .placeholder=${this.hass!.localize(
```

File: src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts [hunk 2]
```diff
@@ -75,6 +78,15 @@ export class ZHAAddGroupPage extends LitElement {
             )}
           ></paper-input>
 
+          <paper-input
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts:81]: We should ideally not add new `paper-input` element...

File: src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts [hunk 3]
```diff
@@ -30,6 +31,8 @@ export class ZHAAddGroupPage extends LitElement {
 
   @state() private _groupName = "";
 
+  @state() private _groupId?: string | number;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts:34]:
```suggestion
  @state() private _groupId?: string;
```

File: src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ import "../../../ha-config-section";
 import "../../../../../components/ha-textfield";
 import "./zha-device-endpoint-data-table";
 import type { ZHADeviceEndpointDataTable } from "./zha-device-endpoint-data-table";
+import { ChangeEvent } from "./types";
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-add-group-page.ts:20]:
```suggestion
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**